public/app/pages/TagExplorerView.tsx

```diff
@@ -193,10 +193,10 @@ function TagExplorerView() {
       return () => fetchData.abort('cancel');
     }
     return undefined;
-  }, [from, until, query, groupByTagValue, dispatch]);
+  }, [from, until, query, groupByTagValue, refreshToken, dispatch]);
 
   useEffect(() => {
-    if (from && until && query) {
+    if (from && until && query && groupByTag) {
```

Comment:
Waits for `groupByTag` to be defined before trying to fetch, avoiding a redundant fetch when this is undefined.